panaroo/integrate.py

```diff
@@ -0,0 +1,272 @@
+            " in the presence/absence csv file (default=max(2, 0.01*n_samples))"
+        ),
+        default=2,
+        type=int) 
+
+    # MSA options
+    core = parser.add_argument_group('Gene alignment')
+    core.add_argument(
+        "-a",
+        "--alignment",
+        dest="aln",
+        help=("Output alignments of core genes or all genes. Options are" +
+              " 'core' and 'pan'. Default: 'None'"),
+        type=str,
+        choices=['core', 'pan'],
+        default=None) 
+    
+    core.add_argument(
+        "--aligner",
+        dest="alr",
+        help=
+        "Specify an aligner. Options:'prank', 'clustal', and default: 'mafft'",
+        type=str,
+        choices=['prank', 'clustal', 'mafft'],
+        default="mafft") 
+    
+    core.add_argument("--core_threshold",
+                      dest="core",
+                      help="Core-genome sample threshold (default=0.95)",
+                      type=float,
+                      default=0.95) 
+    
+    graph = parser.add_argument_group('Graph correction')
+    
+    graph.add_argument("--all_seq_in_graph",
+                       dest="all_seq_in_graph",
+                       help=("Retains all DNA sequence for each gene cluster in the graph " +
+                             "output. Off by default as it uses a large amount of space."),
+                       action='store_true',
+                       default=False)
+        
+    # Other options
+    parser.add_argument("-t",
+                        "--threads",
+                        dest="n_cpu",
+                        help="number of threads to use (default=1)",
+                        type=int,
+                        default=1) 
+    
+    parser.add_argument("--quiet",
+                        dest="quiet",
+                        help="suppress additional output",
+                        action='store_true',
+                        default=False)
+        
+    parser.add_argument("--version'",
+                        action="version",
+                        version='%(prog)s ' + __version__)
+
+    args = parser.parse_args()
+    return (args)
+
+
+def reformat_network(single_gml, output_dir, isolateName): #Generate network output needs to be reformatted for single gff inputs
+    
+    for adj in single_gml._adj:
+        for x in single_gml._adj[adj]:
+            y = single_gml._adj[adj][x]
+        
+            y.pop('members')
+            zero = {'members': 0}
+            y.update(zero)
+        
+            genomes = {'genomeIDs' : '0'}
+            y.update(genomes)
+        
+    for node in single_gml._node:
+        y = single_gml._node[node]
+        y.pop('members')
+        
+        zero = {'members': 0} #members are assigned intbitset[0]. needs to be 0
+        y.update(zero)
+        
+        rep = "[']"
+        for char in rep:
+            y['centroid'] = (str(y['centroid'])).replace(char, '')
```

Comment:
Might be worth looking at the timing of this, the `re` module may be faster

Reply:
Using `re.sub()`

Reply:
changed to a dictionary and function that replaces all characters in one loop